### PR TITLE
provisioning: mini provisioning unit tests are failing

### PR DESCRIPTION
**Pull request by mssawant**
Due to recent changes to the mini provisioner code corresponding unit
tests are failing.

Solution:
Make test code compatible with the related changes in the mini provisioner.

Signed-off-by: Mandar Sawant <mandar.sawant@seagate.com>

### CODE REVIEW

File: provisioning/miniprov/test/test_cdf.py [hunk 1]
```diff
@@ -363,21 +367,19 @@ def ret_motr_mdvalues(value: str) -> Any:
...
 
-        cdf = CdfGenerator(provider=store,
-                           motr_provider=motr_store)
+        cdf = CdfGenerator(provider=store)
         cdf.utils = utils
         ret = cdf._create_node_descriptions()
 
         self.assertIsInstance(ret, list)
-        self.assertEqual(1, len(ret))
+        self.assertEqual(2, len(ret))
```
Comment (SwapnilGaonkar7) [inline provisioning/miniprov/test/test_cdf.py:375]: Why we are checking if len is 2 even if we provided only 1 node 
`'cluster>storage_set[0]>nodes': ['MACH_ID'],`

Comment (mssawant) [reply provisioning/miniprov/test/test_cdf.py:375]: Yeah,

```
    def _create_node_descriptions(self) -> List[NodeDesc]:
        nodes: List[NodeDesc] = []
        conf = self.provider
        # Skipping for controller and HA pod
        machine_ids = conf.get_machine_ids_for_service(
            Const.SERVICE_MOTR_IO.value)

        machine_ids.extend(conf.get_machine_ids_for_service(
            Const.SERVICE_S3_SERVER.value))

        for machine_id in machine_ids:
            nodes.append(self._create_node(machine_id))
        return nodes
```
If all the components are mentioned on the same node, then above code might return a list with 2 items with duplicate machine ids. I think the list needs to be de-depulicated.

Comment (mssawant) [reply provisioning/miniprov/test/test_cdf.py]: @SwapnilGaonkar7 fixed the duplicate entries in machine_ids list.

**Comment by pavankrishnat** [comment]
@mssawant,
pytest test/ cmd failed with ModuleNotFoundError: No module named 'ldap'
Do we need to separate unitstest as per lnet and libfab transports?

```
ERROR: main (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: main
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 153, in loadTestsFromName
    module = __import__(module_name)
  File "/root/PRhare/provisioning/miniprov/hare_mp/main.py", line 39, in <module>
    from cortx.utils.product_features import unsupported_features
  File "/usr/lib/python3.6/site-packages/cortx/utils/product_features/unsupported_features.py", line 21, in <module>
    from cortx.utils.data.db.db_provider import DataBaseProvider, GeneralConfig
  File "/usr/lib/python3.6/site-packages/cortx/utils/data/db/__init__.py", line 21, in <module>
    from cortx.utils.data.db.openldap import OpenLdap
  File "/usr/lib/python3.6/site-packages/cortx/utils/data/db/openldap/__init__.py", line 18, in <module>
    from cortx.utils.data.db.openldap.storage import OpenLdap
  File "/usr/lib/python3.6/site-packages/cortx/utils/data/db/openldap/storage.py", line 22, in <module>
    import ldap
ModuleNotFoundError: No module named 'ldap'
```

**Comment by mssawant** [comment]
@pavankrishnat, probably, cortx-py-utils is not installed?
We do want to add unittests for libfab and lnet endpoints but not in this PR. Its tricky because we do not start actual motr or consul servers.
This PR will focus only on fixing broken unit tests.

**Comment by vaibhavparatwar** [comment]
@mssawant 1N K8s Deployment is faling.

**Comment by suvratjoshi** [comment]
Hi @mssawant 
As I understand, with these changes, these two tests in dir `provisioning/miniprov/` will pass right?
1. python ./setup.py test
2. pytest test/

**Comment by mssawant** [comment]
@suvratjoshi, 
```
(.py3venv) [520478@ssc-vm-1623:miniprov] python3.6 setup.py test
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
writing hare_mp.egg-info/PKG-INFO
writing dependency_links to hare_mp.egg-info/dependency_links.txt
writing entry points to hare_mp.egg-info/entry_points.txt
writing requirements to hare_mp.egg-info/requires.txt
writing top-level names to hare_mp.egg-info/top_level.txt
reading manifest file 'hare_mp.egg-info/SOURCES.txt'
writing manifest file 'hare_mp.egg-info/SOURCES.txt'
running build_ext
test_empty_source_results_empty (test.test_systemd.TestHaxUnitTrasform) ... ok
test_not_everything_commented (test.test_systemd.TestHaxUnitTrasform) ... ok
test_restart_commented (test.test_systemd.TestHaxUnitTrasform) ... ok
test_invalid_machine_id (test.test_validator.TestValidator) ... ok
test_is_cluster_first_node (test.test_validator.TestValidator) ... ok
test_allowed_failure_generation (test.test_cdf.TestCDF) ... ok
test_both_dix_and_sns_pools_can_exist (test.test_cdf.TestCDF) ... ok
test_disk_refs_can_be_empty (test.test_cdf.TestCDF) ... ok
test_dix_pool_uses_metadata_devices (test.test_cdf.TestCDF) ... ok
test_iface_type_can_be_null (test.test_cdf.TestCDF) ... ok
test_invalid_storage_set_configuration_rejected (test.test_cdf.TestCDF)
This test case checks whether exception will be raise if total ... ok
test_it_works (test.test_cdf.TestCDF) ... ok
test_md_pool_ignored (test.test_cdf.TestCDF) ... ok
test_metadata_is_hardcoded (test.test_cdf.TestCDF) ... ok
test_multiple_nodes_supported (test.test_cdf.TestCDF) ... ok
test_provided_values_respected (test.test_cdf.TestCDF) ... ok
test_template_sane (test.test_cdf.TestCDF) ... ok
test_disks_empty (test.test_cdf.TestTypes) ... ok
test_m0clients (test.test_cdf.TestTypes) ... ok
test_m0server_with_disks (test.test_cdf.TestTypes) ... ok
test_maybe_none (test.test_cdf.TestTypes) ... ok
test_pooldesc_empty (test.test_cdf.TestTypes) ... ok
test_protocol (test.test_cdf.TestTypes) ... ok

----------------------------------------------------------------------
Ran 23 tests in 0.287s

OK
(.py3venv) [520478@ssc-vm-1623:miniprov] 
(.py3venv) [520478@ssc-vm-1623:miniprov] pytest test
===================================================================================== test session starts =====================================================================================platform linux -- Python 3.6.8, pytest-6.2.4, py-1.11.0, pluggy-0.13.1
rootdir: /home/520478/cortx-hare-fork/cortx-hare-fork-new/cortx-hare-1/provisioning/miniprov
plugins: aiohttp-0.3.0, mock-3.6.1, timeout-1.4.2, cov-3.0.0
collected 40 items

test/test_cdf.py ..................                                                                                                                                                     [ 45%] test/test_systemd.py ...                                                                                                                                                                [ 52%] test/test_templates.py .................                                                                                                                                                [ 95%] test/test_validator.py ..                                                                                                                                                               [100%]

===================================================================================== 40 passed in 1.48s ======================================================================================(.py3venv) [520478@ssc-vm-1623:miniprov]
```

File: provisioning/miniprov/test/test_cdf.py [hunk 2]
```diff
@@ -1029,24 +1046,41 @@ def my_get(key: str, recurse: bool = False):
...
+        cdf = CdfGenerator(provider=store)
         cdf.utils = utils
         cdf._get_m0d_per_cvg = Mock(return_value=1)
         ret = cdf._create_node_descriptions()
         self.assertIsInstance(ret, list)
         self.assertEqual(2, len(ret))
-        self.assertEqual(Text('srvnode-1.data.private'), ret[0].hostname)
+        self.assertIn(Text('srvnode-1.data.private'),
+                      [ret[0].hostname, ret[1].hostname])
         self.assertEqual(Text('eth1'), ret[0].data_iface)
```
[the remark this answers is not in the document]
Comment (mssawant) [reply provisioning/miniprov/test/test_cdf.py:1079]: @pavankrishnat, fixed, thanks!

**Review by suvratjoshi — APPROVED** [review]

**Review by pavankrishnat — APPROVED** [review]

**Comment by suvratjoshi** [comment]
As this job related information will be lost in some time, just copying the passed 1N job related details here:
The job link of passed 1 node job (before merging the PR) :

http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/Kubernetes/job/hare/132/

image

It has used this job internally:
http://eos-jenkins.colo.seagate.com/job/Cortx-kubernetes/job/setup-cortx-cluster/1951/

**Pull request merged**